Makefile

```diff
@@ -1,6 +1,22 @@
-all:
-	go build GNU_sort.go
-	./GNU_sort GNU_sort.go
+.PHONY: clean
+
+PATH2MAIN := "./cmd/..."
+
+all: build
 
 clean:
-	rm GNU_sort
\ No newline at end of file
+	rm -f sort *.out
```

Comment:
Could `go clean` help you here?

Reply:
I wanted to use it, but it won't clean the sort binary file, I do not know why...

Reply:
What do you think about this?
```sh
$ cd cmd/sort
$ go build
$ go clean
```
If you want to know what commands `go` executes try to add `-x` or `-n`. Also check `go help build` and `go help clean` for the details.
This is what happens in your case:
```sh
# You're inside the root dire of the project
$ go build -n "./cmd/..." # same as "make build"
...
cd .
/home/dgemoli/dev/tools/golang/1.18.1/sdk/pkg/tool/linux_amd64/link -o $WORK/b001/exe/a.out -importcfg $WORK/b001/importcfg.link -buildmode=exe -buildid=feE-DLgqjdmyC32H_Otb/OQ1SZiYrmXYZgp5aJbed/tcgE22N_Adfj7dsX2XYV/feE-DLgqjdmyC32H_Otb -extld=gcc /home/dgemoli/.cache/go-build/86/86fd967a6830f7d79edde45462e6eba6e74a24d408dae68f1801b0a343658c12-d
/home/dgemoli/dev/tools/golang/1.18.1/sdk/pkg/tool/linux_amd64/buildid -w $WORK/b001/exe/a.out # internal
mv $WORK/b001/exe/a.out sort # Here your binary is placed inside the root dir of your project 
...
$ go clean -n "./cmd/..."
cd /home/dgemoli/dev/projects/Prucek/GNU_go_sort/cmd/sort # sort was placed on the root but It cd into "cmd/sort" when it comes to delete it
rm -f sort sort.exe sort sort.exe sort.test sort.test.exe sort.test sort.test.exe main main.exe
```